src/GraphQL/Validation/Rules/UniqueInputFieldNames.cs

```diff
@@ -22,22 +22,22 @@ public class UniqueInputFieldNames : IValidationRule
         /// <exception cref="UniqueInputFieldNamesError"/>
         public Task<INodeVisitor> ValidateAsync(ValidationContext context)
         {
-            var knownNameStack = new Stack<Dictionary<string, IValue>>();
-            var knownNames = new Dictionary<string, IValue>();
-
-            return new EnterLeaveListener(_ =>
-            {
-                _.Match<ObjectValue>(
-                    enter: objVal =>
+            return new NodeVisitors(
+                new MatchingNodeVisitor<ObjectValue>(
+                    enter: (objVal, context) =>
                     {
-                        knownNameStack.Push(knownNames);
-                        knownNames = new Dictionary<string, IValue>();
+                        var knownNameStack = context.TypeInfo.UniqueInputFieldNames_KnownNameStack ??= new Stack<Dictionary<string, IValue>>();
+
+                        knownNameStack.Push(context.TypeInfo.UniqueInputFieldNames_KnownNames);
```

Comment:
I think now this line pushes `null` instead of empty dictionary.

Reply:
That is correct

Reply:
You know better